redbot/core/utils/__init__.py

```diff
@@ -310,7 +313,7 @@ async def __anext__(self) -> _T:
             self._i = 0
             await asyncio.sleep(self._delay)
         self._i += 1
-        return item
+        return await maybe_coroutine(self._map, item)
```

Comment:
Considering that `await`s give control back to the event loop, I'm not sure if it's that good to await if we don't have to - would it be possible to default `self._map` to `None` and only await if it's not `None`?